htcanalyze/main.py

```diff
@@ -562,20 +568,25 @@ def print_results(htcanalyze: HTCAnalyze,
         else:
             results = htcanalyze.analyzed_summary(log_files)  # else
 
-    # Allow this to happen
+    # Allow to end program without output, if no scripts were analyzed
     if results is None:
         sys.exit(0)
 
-    work_with = results
     # convert result to list, if given as dict
     if isinstance(results, dict):
         work_with = [results]
+    else:
+        work_with = results
 
     # check for ignore values
+    # TODO: I think, "mystery" should be named "result"?
```

Comment:
+1 on the mystery of `mystery`

Reply:
I agree that it should be called `result`, especially since there is no need to rename `results` to `work_with` that I can see.

If I'm somehow wrong and `results` is still used somewhere, requiring the use of `work_with` (or some other, more clear, name), then I recommend _against_ calling it `result`, because that implies that `result` is an item inside `results`, which it then wouldn't be. I strongly suggest renaming it from `mystery` though, because I don't like mysteries in code ;)